chainer/functions/pooling_2d.py

```diff
@@ -352,3 +355,95 @@ def average_pooling_2d(x, ksize, stride=None, pad=0, use_cudnn=True):
+
+    def __init__(self, x_shape, pyramid_height, pooling_class, use_cudnn=True):
+        bottom_c, bottom_h, bottom_w = x_shape
+        self.pyramid_height = pyramid_height
+
+        # create pooling functions for different pyramid levels
+        out_dim = 0
+        self.split_inds = []
+        self.poolers = []
+        for pyramid_level in six.moves.range(pyramid_height):
+            num_bins = int(2 ** pyramid_level)
+
+            ksize_h = int(numpy.ceil(bottom_h / (float(num_bins))))
+            remainder_h = ksize_h * num_bins - bottom_h
+            pad_h = remainder_h // 2
+
+            ksize_w = int(numpy.ceil(bottom_w / (float(num_bins))))
+            remainder_w = ksize_w * num_bins - bottom_w
+            pad_w = remainder_w // 2
+
+            ksize = (ksize_h, ksize_w)
+            pad = (pad_h, pad_w)
+
+            if pooling_class == MaxPooling2D:
+                pooler = pooling_class(ksize=ksize, stride=None, pad=pad,
+                                       cover_all=True, use_cudnn=use_cudnn)
+                self.poolers.append(pooler)
+            else:
+                raise NotImplementedError()
+
+            out_dim += bottom_c * (num_bins ** 2)
+            if pyramid_level < pyramid_height - 1:
+                self.split_inds.append(out_dim)
+
+    def forward(self, x):
+        self.ys = []
+        for pooler in self.poolers:
+            y = pooler.forward(x)[0]
+            n, c, h, w = pooler.out_shape = y.shape
+            self.ys.append(y.reshape((n, c * h * w, 1, 1)))
+
+        return concat.Concat(axis=1).forward(self.ys)
+
+    def backward(self, x, gy):
+        if isinstance(x[0], cuda.GPUArray):
+            gx = cuda.zeros_like(x[0])
+        else:
+            gx = numpy.zeros_like(x[0])
+        gys = split_axis.SplitAxis(self.split_inds, axis=1).forward(gy)
+        for pooler, gy in zip(self.poolers, gys):
+            gy = gy.reshape(pooler.out_shape)
+            gx += pooler.backward(x, (gy,))[0]
+
+        return gx,
+
+
+def spatial_pyramid_pooling_2d(x, pyramid_height, pooling_class,
+                               use_cudnn=True):
+    """Spatial pyramid pooling function.
+
+    It outputs a fixed-length vector regardless of input feature map size.
+
+    See detail in paper: `Spatial Pyramid Pooling in Deep Convolutional \
+    Networks for Visual Recognition \
+    <http://arxiv.org/abs/1406.4729>`_.
+
+    Args:
+        x (~chainer.Variable): Input variable. The shape of `x` should be
+            (batchsize, # of channels, height, width).
+        pyramid_height (int): the number of pyramid levels
+        pooling_class (~functions.MaxPooling2D or ~functions.AveragePooling2D):
+            Only MaxPooling2D class can be available for now.
+        use_cudnn (bool): If True and CuDNN is enabled, then this function
+            uses CuDNN as the core implementation.
+
+    Returns:
+        ~chainer.Variable: Ouptut variable.
```

Comment:
Could you write the shape of output variable? Especially the information about `output_variable.shape[1]` would be helpful.